compiler-core/src/parse.rs

```diff
@@ -241,11 +241,15 @@ where
         parse_result: Result<A, ParseError>,
     ) -> Result<A, ParseError> {
         let parse_result = self.ensure_no_errors(parse_result)?;
-        if let Some((start, _, end)) = self.next_tok() {
+        if let Some((start, tok, end)) = self.next_tok() {
             // there are still more tokens
             let expected = vec!["An import, const, type, or function.".into()];
             return parse_error(
                 ParseErrorType::UnexpectedToken {
+                    found: UnexpectedTokenInfo {
+                        display: tok.to_string(),
```

Comment:
Use an enum rather than a string here. It could be the token. I would probably put the entire token as `found` and not have this struct. The `is_reserved_word` can be moved to be a method on Token